### PR TITLE
Fix Date Formatting

### LINKED ISSUE #161: [PE-D][Tester C] editOrder with an invalid date does not cause an error message to appear
The command entered is `editOrder 1 by/30-02-2024 10:10 c/40 d/1xRoses s/PENDING`.
`30-02-2024` is an invalid date, but there is no error message.
The command executes with another date of `29-02-2024` instead, which is unexpected behaviour.

Expected: An error message is shown, indicating that the date is invalid.
Actual: The command executes successfully with a different date.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: aureliony/ped#9

### LINKED ISSUE #165: [PE-D][Tester A] Orders deadline can be set to be before date added
## Problem
The deadline can be set to be earlier than the date added. 
This can be slightly confusing especially with 2 dates so close to each other.

## Editing Order deadline

Deadline is now set to be before the date added.

## Also for ```order```

```order 1 d/1xRoses c/40 by/01-01-1970 00:00```

## Recommended Fix
Maybe you guys could consider having additional checks to have the deadline be after the current time so it's more useful.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: CaptainKeqing/ped#2

### LINKED ISSUE #170: [PE-D][Tester A] Uncaught IllegalArgumentException in order command
## Problem

For your order command, it seems that there are many uncaught IllegalArgumentExceptions.

This causes there to be no feedback given to the user that they have entered something wrong. The only feedback is shown in the terminal used to run the jar file which the user probably won't notice.

### Command: ```order 1 d/1xRoses c/-1 by/23-07-2024 00:00```
*Problem with it: Negative Price
*Expected feedback: Error message about invalid price in feedback box in the app
*Actual feedback: None. Only in terminal

### Command: ```order 1 d/1xRoses c/1 by/40-07-2024 00:00```
*Problem with it: Invalid Day
*Expected feedback: Error message about invalid day in feedback box in the app
*Actual feedback: None. Only in terminal

### Command: ```order 1 d/1xRoses c/1 by/23-07-2024```
*Problem with it: Missing time fields
*Expected feedback: Error message about invalid datetime in feedback box in the app
*Actual feedback: None. Only in terminal

### Command: ```order 1 d/ c/40 by/23-07-2024 00:00```
*Problem with it: Blank description
*Expected feedback: Error message about blank description in feedback box in the app
*Actual feedback: None. Only in terminal

## Recommended Fix
Perhaps it would be good to throw error messages here ! Need to add exception checks to the code to account for these invalid input cases.

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: CaptainKeqing/ped#3

### LINKED ISSUE #177: [PE-D][Tester B] No warning messages upon the incorrect typing of data and time for add order
With the `order` command, users are expected to provide the compulsory field of the deadline of flower delivery. However, there is no warning or error message upon the typo of a wrong date or time which cannot exist. Nothing happens and the user might be confused as to why the function was not performing as expected. Referring to the command below:

I personally thought that it would be good to have a reminder instead of doing nothing.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Jayne1010/ped#4

### LINKED ISSUE #179: [PE-D][Tester D] No error message is displayed when the date time format is invalid when we are adding an order.
https://youtu.be/sLdMDBnXSBk

As seen in the clip, there is no error message displayed by the application

Expected: Error message shown :Invalid datetime format 
Actual: No error displayed

Steps to reproduce
order command with an invalid datetime.

Suggestion: Display an error message when the datetime format is invalid.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: shamesjen/ped#9

### LINKED ISSUE #198: [PE-D][Tester B] Deadline field allows for invalid date to be added.
During the addition of a new order, the deadline field allows for invalid date and the addition of the order will still be successful. For example, `01-01-0001` is allowed, but an error message should have been thrown instead. 

In the current real world context, dates such as `01-01-0001` does not exist, as such it would be good to have a reminder message appear when the user tries to input dates like so.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Jayne1010/ped#10

### PR DESCRIPTION
**Pull request by Jaspertzx**
Users can enter an invalid date, and no error message would show up.

This makes it confusing as they are left wondering why their commands is invalid.

Fix the date formatting to throw an error message on the output box for the users 
to see by ensuring that commands are using the parser method.

This will allow users to see the error.

Resolves #161
Resolves #165
Resolves #177
Resolves #170
Resolves #179
Resolves #198

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -156,7 +156,11 @@ public static Deadline parseDeadline(String deadline) throws ParseException {
         if (!Deadline.isValidDeadline(trimmedDeadline)) {
             throw new ParseException(Deadline.MESSAGE_CONSTRAINTS);
         } else {
-            return new Deadline(trimmedDeadline);
+            Deadline deadlineObj = new Deadline(trimmedDeadline);
+            if (!Deadline.isAfterCurrentDate(deadlineObj)) {
+                throw new ParseException(Deadline.INVALID_DATE_CONSTRAINTS);
+            }
+            return deadlineObj;
```
Comment (rertyy) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:163]: Nested

File: src/main/java/seedu/address/model/order/Deadline.java [hunk 2]
```diff
@@ -17,6 +17,9 @@ public class Deadline implements Comparable<Deadline> {
             "A deadline should be in the format of "
                     + "DD-MM-YYYY HH:MM, e.g. 01-01-2024 23:59";
 
+    public static final String INVALID_DATE_CONSTRAINTS = "The deadline provided must be set for a time in the future "
+            + "and must not be before the current time.";
```
Comment (rertyy) [inline src/main/java/seedu/address/model/order/Deadline.java:21]: Remove this and put in UG

Comment (rertyy) [reply src/main/java/seedu/address/model/order/Deadline.java:21]: Silent error

**Comment by Jaspertzx** [comment]
https://github.com/nus-cs2103-AY2324S2/forum/issues/760

**Comment by rertyy** [comment]
To potenttially put under planned enhancements

**Pull request closed without merging**